src/content/docs/configuration/v4_migration.md

```diff
@@ -3,132 +3,143 @@ id: v4_migration
+📖 `:help astrocore`
 
-- `lsp.server-settings` has been renamed to `lsp.config`. If you have the `["server-settings"]` table in your `user/init.lua` file, just rename it to `config`. If you have the folder `user/lsp/server-settings`, just rename the folder to `user/lsp/config`.
+The following version 3 `user/init.lua` configuration keys move to the same keys in the version 4 `lua/plugins/astrocore.lua` template file:
 
-- `luasnip` options table has been removed. Please see the updated [Custom Snippets Documentation](../../recipes/snippets) for the new way to extend the default configuration of LuaSnip to add new loaders.
+- `autocmds`
+- `mappings`
+- `options`
 
-- `which-key` options table has been removed. Which-key menu titles can now be easily added into the `mappings` table by setting a binding to a table with the `name` property set and it will be passed to `which-key`. For example, you can add the following to the `mappings` table: `["<Leader>b"] = { name = "Buffer" }`.
+Additional non version 3 keys configured in the `lua/plugins/astrocore.lua` file are:
 
-- `nvim-autopairs` options table has been removed. Please see the updated [Customize Autopairs Documentation](../../recipes/autopairs) for the new way to extend the default configuration of autopairs and adding more rules.
+- `commands`
+- `features`
+- `git_worktrees`
+- `on_keys`
+- `rooter`
+- `sessions`
 
-- `cmp` options table has been removed. Please see the updated [Customize cmp Completion Documentation](../../recipes/cmp) for the new way to extend the default configuration of cmp and running more `cmp` setup functions.
+### Migrating to AstroLSP
 
-- `mason-lspconfig`, `mason-null-ls`, and `mason-nvim-dap` options tables have been removed, please use the new plugin notation for extending these options like adding custom setup handlers. This is described in the [Extending Core Plugin Config Functions Documentation](../custom_plugins#extending-core-plugin-config-functions).
+🔗 [AstroLSP example configuration](https://github.com/AstroNvim/astrolsp/tree/main#%EF%B8%8F-configuration)
+📖 `:help astrolsp`
 
-- `default_theme` has been migrated to a dedicated plugin that can be used outside of AstroNvim as well at [AstroNvim/astrotheme](https://github.com/AstroNvim/astrotheme). This can be customized and configured the same as any other plugin, check the README for details on the `setup` function.
+The following version 3 `user/init.lua` configuration keys move to the same keys in the version 4 `lua/plugins/astrolsp.lua` template file:
 
-- The bindings in `cmp` to scroll the preview window for a completion item have moved to `<c-u>` and `<c-d>`
+- `diagnostics`
+- `lsp.capabilities`
+- `lsp.config.<lsp>`
+- `lsp.flags`
+- `lsp.formatting`
+- `lsp.mappings`
+- `lsp.on_attach`
+- `lsp.servers`
+- `lsp.setup_handlers` ?? Where does this go? I'm guessing it is `config.<lang>`
+- `lsp.skip_setup` ?? Where does this go? I'm guessing it is `config.<lang>`
```

Comment:
This is handled in AstroLSP with the `handlers` table by simply setting the handler for a language server to `false`